pyOCD/flash/flash.py

```diff
@@ -105,20 +110,42 @@ def __init__(self, target, flash_algo):
     def minimumProgramLength(self):
         return self.min_program_length
 
-    def init(self):
+    def init(self, adr=None, clk=0, fnc=None):
```

Comment:
these init uninit functions look as a best match for context management using the `with` clause